src/main/java/org/apache/maven/resolver/internal/ant/tasks/CreatePom.java

```diff
@@ -0,0 +1,345 @@
+        this.licenses = licenses;
+    }
+
+    /**
+     * Add a repositories element to the POM.
+     * This allows you to specify one or more repositories where dependencies can be found. Note that
+     * specifying repositories in the POM is discouraged if you aim to publish to maven central.
+     *
+     * @param repositories the repositories to add to the POM
+     */
+    public void addRepositories(Repositories repositories) {
+        this.repositories = repositories;
+    }
+
+    /**
+     * Execute the task to create the POM file.
+     * This method will create the POM file with the specified properties and dependencies.
+     */
+    @Override
+    public void execute() {
+        if (!pomFile.getParentFile().exists()) {
+            pomFile.getParentFile().mkdirs();
+        }
+        MavenProject pom = new MavenProject();
+        pom.setGroupId(getGroupId());
+        pom.setArtifactId(getArtifactId());
+        pom.setVersion(getVersion());
+        pom.setName(getName());
+        pom.setDescription(getDescription());
+
+        if (dependencyManagementRef != null) {
+            DependencyManagement dependencyManagement = DependencyManagement.get(getProject(), dependencyManagementRef);
+            appendManagedDependencies(dependencyManagement.getDependencies(), pom);
+        }
+        if (dependenciesRef != null) {
+            Dependencies dependencies = new Dependencies();
+            dependencies.setProject(getProject());
+            org.apache.tools.ant.types.Reference ref =
+                    new org.apache.tools.ant.types.Reference(getProject(), dependenciesRef);
+            dependencies.setRefid(ref);
+
+            appendDependencies(dependencies, pom);
+        }
+
+        if (licenses != null) {
+            for (License l : licenses.getLicenses()) {
+                pom.addLicense(l);
+            }
+        }
+
+        if (repositories != null) {
+            repositories.getRepositories().forEach(repository -> {
+                org.apache.maven.model.Repository repo = new org.apache.maven.model.Repository();
+                repo.setId(repository.getId());
+                repo.setName(repository.getName());
+                repo.setUrl(repository.getUrl());
+                repo.setLayout(repository.getLayout());
+                if (repository.getReleases() != null) {
+                    RepositoryPolicy policy = new RepositoryPolicy();
+                    policy.setEnabled(repository.getReleases().isEnabled());
+                    repo.setReleases(policy);
+                }
+                if (repository.getSnapshots() != null) {
+                    RepositoryPolicy policy = new RepositoryPolicy();
+                    policy.setEnabled(repository.getSnapshots().isEnabled());
+                    repo.setSnapshots(policy);
+                }
+                pom.getModel().getRepositories().add(repo);
+            });
+        }
+
+        try (FileWriter fw = new FileWriter(pomFile)) {
```

Comment:
This must have an encoding, preferably UTF-8. If you can't do it with a FileWriter do it with a FileOutputStream chained to an OutputStreamWriter.

Reply:
OK, changed to:
try (Writer writer = new OutputStreamWriter(Files.newOutputStream(pomFile.toPath()), StandardCharsets.UTF_8))